src/Stratis.Bitcoin.Features.BlockStore/BlockStoreSignaled.cs

```diff
@@ -149,26 +152,14 @@ public void RelayWorker()
                 {
                     this.logger.LogTrace("Checking if block '{0}' is on disk.", block);
 
-                    // The first block that is not on disk will abort the loop.
-                    if (!await this.blockRepository.ExistAsync(block.HashBlock).ConfigureAwait(false))
```

Comment:
@noescape00 I tried to follow as much of the discussion as possible in the issue thread... So here we don't need to check whether or not the blocks exist on disk anymore? Why is this?

Reply:
The reason for checking was that we were not sure if the block was saved already or not. If we use cache we don't need to wait for it to be persisted on the disk so we can skip that step and only verify if the block is still a part of the best chain we're aware of.

Reply:
Ah OK I understand :)

Reply:
This is mainly done so peers that receive the block being broadcast can still ask for the block in the cache even if store loop was to slow to push to the repo.